walk/walk.go

```diff
@@ -351,6 +345,31 @@ func (r *symlinkResolver) follow(c *config.Config, dir, rel, base string) bool {
 			return false
 		}
 	}
+
+	// Ignore symlinks into Bazel's output.
+	// This includes the bazel-* convenience symlinks and any created for IDEs.
+	// Start by getting a canonical path to Bazel's output base.
+	// Note that we expect the bazel-out symlink in the workspace root directory to point to
+	// <output-base>/execroot/<workspace-name>/bazel-out
+	// See https://docs.bazel.build/versions/master/output_directories.html
+	// for documentation on Bazel directory layout.
+	bazelOut, err := filepath.EvalSymlinks(filepath.Join(c.RepoRoot, "bazel-out"))
```

Comment:
This introduces a new dependency on the value of `--symlink_prefix` and also means that symlinks into the output base are still followed if the convenience symlinks aren't created (which is more likely with the new algorithm used in Bazel 6).

@cpsauer @linzhp Do you think we could instead decline to follow any symlinks that point out of the project directory? That is a condition that needs no external information to be evaluated.

Reply:
Yeah, I think that's a fair requirement